### PR TITLE
refactor CMakeLists

### PR DESCRIPTION
**Pull request by okdshin**
- remove un target_xxx commands
- move MKLDNN setup to menoh/CMakeLists.txt
- add git finder
- remove `glob` for `SRC`

### CODE REVIEW

**Comment by msakai** [comment]
`libmenoh.so` is now linked with `libonnx.so` and `libonnx_proto.so`. Is this an intended change?

Also, it fails to compile with MSVC.
cc: @durswd

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -45,4 +48,4 @@ if(UNIX AND NOT APPLE)
             LINK_FLAGS "-Wl,--disable-new-dtags")
 endif()
 
-add_test(NAME menoh_test COMMAND menoh_test)
+#add_test(NAME menoh_test COMMAND menoh_test)
```
Comment (msakai) [inline test/CMakeLists.txt]: Why `enable_testing()` is removed and `add_test()` is commented out?

Comment (okdshin) [reply test/CMakeLists.txt]: It didn't worked before. I fixed it by new commit. Please check.

Comment (msakai) [reply test/CMakeLists.txt]: Thanks. I checked that `ctest` works.

Comment (okdshin) [reply test/CMakeLists.txt]: Thanks! Do you have any other review comments?

**Pull request merged**